frontend/src/components/table/schedule/use-schedule-state.ts

```diff
@@ -0,0 +1,60 @@
+import { Dispatch, useReducer, useState } from "react";
+import { ScheduleLogic } from "../../../logic/real-schedule-logic/schedule.logic";
+import { ActionModel } from "../../../state/models/action.model";
+import { ScheduleDataModel } from "../../../state/models/schedule-data/schedule-data.model";
+import {
+  ScheduleActionType,
+  ScheduleComponentState,
+  scheduleInitialState,
+} from "./schedule-state.model";
+
+export function useScheduleState(): [
+  ScheduleComponentState,
+  Dispatch<ActionModel<ScheduleComponentState>>,
+  (scheduleData: ScheduleDataModel) => void,
+  ScheduleLogic?
+] {
+  const [scheduleLogic, setScheduleLogic] = useState<ScheduleLogic>();
+  const [scheduleState, dispatchScheduleState] = useReducer(
+    (state: ScheduleComponentState, action: ActionModel<ScheduleComponentState>) => {
+      let data;
+      switch (action.type) {
```

Comment:
Pewnie dało by się to zapisać jakoś krócej, ale ja na razie nie wykliniłem.
@pectom @Sig00rd  jak macie jakieś pomysły - to piszcie